core/src/main/scala/kafka/consumer/ZookeeperConsumerConnector.scala

```diff
@@ -588,7 +588,10 @@ private[kafka] class ZookeeperConsumerConnector(val config: ConsumerConfig,
             if (doRebalance)
               syncedRebalance
           } catch {
-            case t: Throwable => error("error during syncedRebalance", t)
+            case t: Throwable => {
+              error("error during syncedRebalance", t)
+              shutdown()
```

Comment:
I agree that it is a weird state to be in and shutting down is not unreasonable. However, this is a major change in behavior especially given the decentralized coordination in the old consumer. The main issue that I see is that one misbehaving client that is slow to release partitions can cause the good clients to shutdown.

Specifically at LinkedIn, one common occurrence of rebalance failures is when a developer starts up a consumer on his development machine that joins a group in a staging environment (i.e., in a remote DC) - sure developers should use a different group but for various reasons this happens. This actually causes rebalance failures in the staging instances. If those instances were to shutdown the staging services would need to be bounced (and some of them may even die depending on how they handle the consumer's lifecycle).

So it seems at the very least we should make such a change configurable and default to the old behavior.